### PR TITLE
test: Enhance test coverage for KleinanzeigenBot initialization and core functionality

### PR DESCRIPTION
**Pull request by 1cu**
## ℹ️ Description
This PR significantly enhances the test coverage of the KleinanzeigenBot class by adding comprehensive test suites in test_init.py. The changes introduce structured test organization, reusable fixtures, and thorough testing of core functionalities.

- Link to the related issue(s): none
- The changes focus on improving test coverage and maintainability while ensuring proper type checking and development tooling.

### Test Improvements (test_init.py)
- Added test classes for distinct functionality areas:
  - `TestKleinanzeigenBotInitialization`
  - `TestKleinanzeigenBotLogging`
  - `TestKleinanzeigenBotCommandLine`
  - `TestKleinanzeigenBotAuthentication`
  - `TestKleinanzeigenBotLocalization`
  - `TestKleinanzeigenBotBasics`
  - `TestKleinanzeigenBotArgParsing`
  - `TestKleinanzeigenBotCommands`
  - `TestKleinanzeigenBotAdOperations`
  - `TestKleinanzeigenBotAdManagement`
  - `TestKleinanzeigenBotAdConfiguration`
  - `TestKleinanzeigenBotAdDeletion`
  - `TestKleinanzeigenBotAdRepublication`
  - `TestKleinanzeigenBotShippingOptions`
  - `TestKleinanzeigenBotUrlConstruction`

- Added key fixtures:
  - `mock_page`: Provides a mock page object for testing
  - `base_ad_config`: Provides a base ad configuration
  - `minimal_ad_config`: Provides a minimal ad configuration

- Added helper functions:
  - `create_ad_config`: Creates new ad configurations
  - `remove_fields`: Removes specified fields from configurations

### Configuration and Dependency Changes
- Added types-PyYAML (6.0.12.20241230) for improved type checking
- Updated pyproject.toml:
  - Added pylint configuration for test files
  - Added types-PyYAML to dev dependencies
- Updated pdm.lock:
  - Removed redundant test group
  - Updated content hash
  - Added types-PyYAML package

### ⚙️ Type of Change
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (adds new functionality without breaking existing usage)

## ✅ Checklist
- [x] I have reviewed my changes to ensure they meet the project's standards
- [x] I have tested my changes and ensured that all tests pass (`pdm run test`)
- [x] I have verified that linting passes (`pdm run lint`)
- [x] I have run security scans and addressed any identified issues (`pdm run audit`)
- [x] I have updated documentation where necessary

## 📝 Additional Notes
- The test improvements provide better coverage for edge cases and error conditions
- The addition of types-PyYAML improves development experience with better IDE support
- Test organization follows best practices with clear separation of concerns
- All changes maintain backward compatibility

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -246,4 +252,5 @@ dev = [
     "mypy",
     "pyright",
     "pytest-cov>=6.0.0",
+    "types-PyYAML"  # Type stubs for PyYAML
```
Comment (sebthom) [inline pyproject.toml]: since we are using ruamel.yaml and not PyYAML, how does this help?

Comment (sebthom) [reply pyproject.toml]: Btw. there is a pending PR to support ruamel.yaml https://github.com/python/typeshed/pull/12584

File: tests/unit/test_init.py [hunk 2]
```diff
@@ -0,0 +1,1013 @@
+"""
+SPDX-FileCopyrightText: © Sebastian Thomschke and contributors
```
Comment (sebthom) [inline tests/unit/test_init.py]: When adding new files with new code please either use your name in the copyright header or use something like "SPDX-FileCopyrightText: © The kleinanzeigen-bot authors and contributors"

**Comment by sebthom** [comment]
@1cu do you think you can come up with a way that the actual test coverage of a PR is added as comment to the PR? Using something like https://github.com/marketplace/actions/code-coverage-summary or similar.
